viewing_party/party.py

```diff
@@ -1,23 +1,196 @@
 # ------------- WAVE 2 --------------------
 # -----------------------------------------
+def get_watched_avg_rating(user_data):
+    if user_data["watched"] == []: # watched list of movies is empty
+            return 0.0
+    average_rating = [] #list to keep rating of each movie
+
+    # add rating of each movie to "average_rating" list
+    # loop go through list of movies in watched list
+    for movie in user_data["watched"]:
+        average_rating.append(movie["rating"])
+    return sum(average_rating) / len(average_rating)
+
+
+def get_most_watched_genre(user_data):
+    if user_data["watched"] == []: # empty user's watched list
+            return None
+    
+    genre_dict = {} # dictionary to keep genre and it's frequentcy, example {"fantasy": 2, "horror": 1, "action":5}
+
+    #1 get all genres and their frequency from user_data
+    for movie in user_data["watched"]:
+        if movie["genre"] not in genre_dict: # if genre is not yet in dictionary, set its frequency to 1
+            genre_dict[movie["genre"]] = 1
+        else:
+            genre_dict[movie["genre"]] += 1 # if genre is already in dictionary, increase frequency by 1
+
+    #2 get the key with max value
+    for genre in genre_dict:
+        if genre_dict[genre] == max(genre_dict.values()):
+            return genre 
+    # max_rating = max(genre_dict.values())
+    # most_genre = list(genre_dict.keys())[list(genre_dict.values()).index(max_rating)]
+    # return most_genre
+
+    
+    
+
 
 
 # -----------------------------------------
 # ------------- WAVE 3 --------------------
 # -----------------------------------------
 
+def get_unique_watched(user_data):
+    friends_watched = []
+    unique_watched_user = []
+
+    #get movies friends watched
+    # outer loop goes through list of friends of user
+    for friend in user_data["friends"]:
+        # inner loop goes through list of friend's watched movies 
+        for friend_movie in friend["watched"]:
+            friends_watched.append(friend_movie)
+
+    # get unique movies for user (that movies friends didn't watch)
+    #add user movie to list of dictionaries if user's watched title not in the list of friend watched titles
+    for user_movie in user_data['watched']:
+        if user_movie not in friends_watched:
+            unique_watched_user.append(user_movie)
+    
+    return unique_watched_user
+
+
+def get_friends_unique_watched(user_data):
+    user_watched = []
+    friends_unique_movies = []
+    # get the list of movies user watched
+    for movie in user_data["watched"]:
+        user_watched.append(movie)
+
+    # get the list of movies friends watched but user not
+    # outer loop goes through list of friends of user
+    for friend in user_data["friends"]:
+        # inner loop goes through list of friend's watched movies
+        for friend_movie in friend["watched"]:
+            if friend_movie not in user_watched:
+                if friend_movie not in friends_unique_movies:

```

Comment:
We could combine these two if statements into a compound conditional

```suggestion
            if friend_movie not in user_watched and friend_movie not in friends_unique_movies:
```